app/graphql/mutations/update_sample_metadata.rb

```diff
@@ -13,22 +13,67 @@ class UpdateSampleMetadata < BaseMutation
              description: 'Persistent Unique Identifier of the sample. For example, `INXT_SAM_AAAAAAAAAA`.'
     validates required: { one_of: %i[sample_id sample_puid] }
 
-    field :errors, [String], null: false, description: 'A list of errors that prevented the mutation.'
-    field :sample, Types::SampleType, null: false, description: 'The updated sample.'
-    field :status, GraphQL::Types::JSON, null: false, description: 'The status of the mutation.'
+    field :errors, [Types::UserErrorType], description: 'A list of errors that prevented the mutation.'
+    field :sample, Types::SampleType, null: true, description: 'The updated sample.'
+    field :status, GraphQL::Types::JSON, null: true, description: 'The status of the mutation.'
 
-    def resolve(args)
+    def resolve(args) # rubocop:disable Metrics/MethodLength,Metrics/AbcSize
       sample = if args[:sample_id]
                  IridaSchema.object_from_id(args[:sample_id], { expected_type: Sample })
                else
                  Sample.find_by(puid: args[:sample_puid])
                end
+      if sample.nil?
```

Comment:
**nitpick (blocking):** Same comment as in app/graphql/mutations/attach_files_to_sample.rb